Content.Server/Traitor/Uplink/UplinkSystem.cs

```diff
@@ -206,9 +206,9 @@ private void UpdateUserInterface(UplinkComponent component)
                                     break;
                                 }
                             }
-                            if (!found)
-                                listings.Remove(entry);
                         }
+                        if (!found)
+                            listings.Remove(entry);
```

Comment:
i needs to get decremented. Also RemvoeSwap is slightly faster.
```suggestion
                        if (!found)
                        {
                            listings.RemoveSwap(i);
                            i--;
                        }
```